src/core/src/op/util/gather_base.cpp

```diff
@@ -5,145 +5,40 @@
-    OPENVINO_SUPPRESS_DEPRECATED_END
-    OPENVINO_ASSERT(const_op, "axis value is not set");
-
-    int64_t axis = const_op->cast_vector<int64_t>()[0];
-    if (axis < 0) {
-        const auto& data_rank = get_input_partial_shape(0).rank();
-        if (data_rank.is_static()) {
-            axis += data_rank.get_length();
-        }
-    }
-    return axis;
-}
-
-const int64_t& ov::op::util::GatherBase::get_batch_dims() const {
-    return m_batch_dims;
-}
-
-void ov::op::util::GatherBase::set_batch_dims(int64_t batch_dims) {
-    m_batch_dims = batch_dims;
-}
-
-OPENVINO_SUPPRESS_DEPRECATED_START
+namespace ov {
+namespace op {
 namespace gather {
 namespace {
-template <ov::element::Type_t ET>
-bool evaluate(const ngraph::HostTensorPtr& arg0,
-              const ngraph::HostTensorPtr& arg1,
-              const ngraph::HostTensorPtr& out,
-              int64_t axis,
-              int64_t batch_dims) {
-    using T = typename ov::element_type_traits<ET>::value_type;
-    ov::Shape params_shape = arg0->get_shape();
-    ov::Shape indices_shape = arg1->get_shape();
-    ov::Shape out_shape(params_shape.size() + indices_shape.size() - 1 - batch_dims);
-    int64_t i = 0;
-    for (; i < axis; i++) {
-        out_shape[i] = params_shape[i];
-    }
-    for (int64_t j = batch_dims; j < static_cast<int64_t>(indices_shape.size()); i++, j++) {
-        out_shape[i] = indices_shape[j];
-    }
-    for (int64_t j = axis + 1; j < static_cast<int64_t>(params_shape.size()); i++, j++) {
-        out_shape[i] = params_shape[j];
-    }
-
-    out->set_shape(out_shape);
-
-    if (arg1->get_element_type() == ov::element::i64) {
-        ov::reference::gather<T, int64_t>(arg0->get_data_ptr<ET>(),
-                                          arg1->get_data_ptr<int64_t>(),
-                                          out->get_data_ptr<ET>(),
-                                          arg0->get_shape(),
-                                          arg1->get_shape(),
-                                          out->get_shape(),
-                                          axis,
-                                          batch_dims);
-    } else if (arg1->get_element_type() == ov::element::i32) {
-        ov::reference::gather<T, int32_t>(arg0->get_data_ptr<ET>(),
-                                          arg1->get_data_ptr<int32_t>(),
-                                          out->get_data_ptr<ET>(),
-                                          arg0->get_shape(),
-                                          arg1->get_shape(),
-                                          out->get_shape(),
-                                          axis,
-                                          batch_dims);
-    } else {
-        OPENVINO_THROW("Unexpected type ", arg1->get_element_type().c_type_string(), " for Gather evaluate method.");
-    }
 
-    return true;
-}
+Shape out_shape_infer(const Shape& data_shape, const Shape& indices_shape, int64_t axis, int64_t batch_dims) {
```

Comment:
why not use `shape_infer()`?

Reply:
Is just complex as shapes will be converted to PartialShape, there is a lot of check which are not required.
As this implementation worked and shapes are correct to make correct calculations be reference implementation I decide to keep it and re-factored to use `std` functions. 

As future improvement this implementation could be merged with shape_infer to give required validation and simplicity of calculation, but didn't want to change it in this PR